### PR TITLE
Add support for skunk

### PR DESCRIPTION
**Pull request by jnicoulaud-ledger**
Adds support for [Skunk](https://typelevel.org/skunk).

### CODE REVIEW

**Comment by jnicoulaud-ledger** [comment]
Not sure this is the good way of doing things, if I try the case class example with implicit conversions:
```scala
// defining a codec for a refined case class
final case class User(name: Username, age: Int :| Positive)
given Codec[User] = (varchar *: int4).to[User]
```

Compilation fails with a compiler crash:
```
java.lang.AssertionError: assertion failed while typechecking /home/jnicoulaud/work/foss/iron/skunk/test/src/io/github/iltotore/iron/SkunkExample.scala
```

**Comment by Iltotore** [comment]
What is the full error thrown by the compiler?

**Comment by jnicoulaud-ledger** [comment]
https://gist.github.com/jnicoulaud-ledger/4de31bd71a8a0562d1fba1c5ab75eaf8

**Comment by Iltotore** [comment]
The test module, including this part:

```scala
// defining a codec for a refined case class
final case class User(name: Username, age: Int :| Positive)
given Codec[User] = (varchar.refined[Not[Blank]] *: int4.refined[Positive]).to[User]
```

seems to compile and work as expected. Am I missing something?

File: skunk/src/io.github.iltotore.iron/skunk.scala [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  extension [A](codec: Codec[A])
+    inline def refined[C](using inline constraint: Constraint[A, C]): Codec[A :| C] =
+      codec.eimap[A :| C](_.refineEither[C])(_.asInstanceOf[A])
+
+  /**
+   * Implicit conversion for refining a [[Codec]]. Decodes to the underlying type then checks the constraint.
+   *
+   * @param constraint the [[Constraint]] implementation to test the decoded value
+   */
+  inline given [A, C](using inline constraint: Constraint[A, C]): Conversion[Codec[A], Codec[A :| C]] =
```
Comment (Iltotore) [inline skunk/src/io.github.iltotore.iron/skunk.scala:23]: I don't think such conversion should be included. It adds magic and complexity. Furthermore, doing `codec.refined[Constraint]` or summoning `Codec[A :| C]` does not seem painful to me.

Also, note that similar Iron modules (ZIO JSON/Circe/Doobie/... support) do not include such constraint.

**Review by Iltotore — CHANGES_REQUESTED** [review]
The PR looks good to me. I just think the introduced implicit conversion is not necessary and causes more issues (e.g the compiler crash) than benefits.

File: skunk/src/io.github.iltotore.iron/skunk.scala (hunk 1 above)
Comment (jnicoulaud-ledger) [reply skunk/src/io.github.iltotore.iron/skunk.scala]: Agreed, I got rid of it (with a few other fixes, please review again)

**Review by Iltotore — APPROVED** [review]

**Pull request merged**